pubsub/integration_test.go

```diff
@@ -1263,23 +1263,18 @@ func TestIntegration_OrderedKeys_JSON(t *testing.T) {
 		key := parts[0]
 		msg := parts[1]
 		publishData = append(publishData, testutil2.OrderedKeyMsg{Key: key, Data: msg})
-		r := topic.Publish(ctx, &Message{
+		topic.Publish(ctx, &Message{
 			Data:        []byte(msg),
 			OrderingKey: key,
 		})
-		go func() {
-			_, err := r.Get(ctx)
-			if err != nil {
-				// Can't fail inside goroutine, so just log the error.
-				t.Logf("publish error for message(%s): %v", msg, err)
-			}
-		}()
 		wg.Add(1)
```

Comment:
The waitgroup pattern here looks a little racy at a glance.  It looks like you can get to zero in the wg multiple times if the publish and subscribes keep up with another, which may fire the done early?  The test doesn't seem to compare it the publish/subscribe counts are aligned.

Reply:
I believe that all of the `wg.Add(1)` calls will always happen before the goroutine that calls `Receive` is executed (where `wg.Done` is called). This should prevent reaching zero in the waitgroup multiple times.

Reply:
Yeah, appears I misread this initially.  Thanks for confirming.